src/client/store/middlewares/async.js

```diff
@@ -0,0 +1,18 @@
+function isPromise(value) {
```

Comment:
I'm not sure I get this part.
Where does action comes from?
if resolve(value) == value you trigger it then dispatch the result as a payload?

Is that the difference between having an action to do and just having to store something?

Reply:
To reproduce Redux and make our own global state manager we need two native features:
- Context
- reducers

In React, reducers can't handle asynchronous actions (e.g.: `async`, `Promise`, `then` are all sorts of asynchronous methods).
So this file expose a middleware that check weither an action from the reducer is async or not.
If it is, it resolve it before send it to the dispatcher.
Else, it continue as if nothing happened